Applications/Spire/Source/Ui/SortedTableModel.cpp

```diff
@@ -0,0 +1,216 @@
+#include "Spire/Ui/SortedTableModel.hpp"
+#include <algorithm>
+#include "Spire/Ui/DecimalBox.hpp"
+
+using namespace boost::posix_time;
+using namespace boost::signals2;
+using namespace Spire;
+
+SortedTableModel::ColumnOrder SortedTableModel::ColumnOrder::cycle() const {
+  if(m_order == Ordering::NONE) {
+    return ColumnOrder{m_index, Ordering::ASCENDING};
+  } else if(m_order == Ordering::ASCENDING) {
+    return ColumnOrder{m_index, Ordering::DESCENDING};
+  } 
+  return ColumnOrder{m_index, Ordering::NONE};
+}
+
+SortedTableModel::SortedTableModel(std::shared_ptr<TableModel> source)
+  : SortedTableModel(source, m_order) {}
+
+SortedTableModel::SortedTableModel(std::shared_ptr<TableModel> source,
+  std::vector<ColumnOrder> order)
+    : SortedTableModel(source, order, get_default_comparator()) {}
+
+SortedTableModel::SortedTableModel(std::shared_ptr<TableModel> source,
+  Comparator comparator)
+    : SortedTableModel(source, m_order, comparator) {}
+
+SortedTableModel::SortedTableModel(std::shared_ptr<TableModel> source,
+  std::vector<ColumnOrder> order, Comparator comparator)
```

Comment:
When a constructor has a body, double indent the parameter and colon, and triple indent additional members in the initializer list.